cli/ballerina-cli/src/main/java/io/ballerina/cli/cmd/DocCommand.java

```diff
@@ -0,0 +1,217 @@
+ *  Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing,
+ *  software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied.  See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ */
+package io.ballerina.cli.cmd;
+
+import io.ballerina.cli.TaskExecutor;
+import io.ballerina.cli.task.CompileTask;
+import io.ballerina.cli.task.CreateDocsTask;
+import io.ballerina.cli.task.CreateTargetDirTask;
+import io.ballerina.projects.Project;
+import io.ballerina.projects.directory.BuildProject;
+import io.ballerina.projects.util.ProjectConstants;
+import org.ballerinalang.compiler.CompilerPhase;
+import org.ballerinalang.docgen.docs.BallerinaDocGenerator;
+import org.ballerinalang.tool.BLauncherCmd;
+import org.wso2.ballerinalang.compiler.util.CompilerContext;
+import org.wso2.ballerinalang.compiler.util.CompilerOptions;
+import org.wso2.ballerinalang.compiler.util.ProjectDirs;
+import picocli.CommandLine;
+
+import java.io.PrintStream;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.List;
+
+import static io.ballerina.cli.cmd.Constants.DOC_COMMAND;
+import static org.ballerinalang.compiler.CompilerOptionName.COMPILER_PHASE;
+import static org.ballerinalang.compiler.CompilerOptionName.EXPERIMENTAL_FEATURES_ENABLED;
+import static org.ballerinalang.compiler.CompilerOptionName.LOCK_ENABLED;
+import static org.ballerinalang.compiler.CompilerOptionName.OFFLINE;
+import static org.ballerinalang.compiler.CompilerOptionName.PRESERVE_WHITESPACE;
+import static org.ballerinalang.compiler.CompilerOptionName.SKIP_TESTS;
+import static org.ballerinalang.compiler.CompilerOptionName.TEST_ENABLED;
+
+/**
+ * This class represents the "ballerina doc" command.
+ *
+ * @since 2.0.0
+ */
+@CommandLine.Command(name = DOC_COMMAND, description = "Ballerina doc - Generates API Documentation")
+public class DocCommand implements BLauncherCmd {
+
+    private final PrintStream outStream;
+    private final PrintStream errStream;
+    private Path projectPath;
+    private boolean exitWhenFinish;
+
+    public DocCommand() {
+        this.projectPath = Paths.get(System.getProperty("user.dir"));
+        this.outStream = System.out;
+        this.errStream = System.err;
+        this.exitWhenFinish = true;
+    }
+
+    public DocCommand(PrintStream outStream, PrintStream errStream, boolean exitWhenFinish) {
+        this.projectPath = Paths.get(System.getProperty("user.dir"));
+        this.outStream = outStream;
+        this.errStream = errStream;
+        this.exitWhenFinish = exitWhenFinish;
+    }
+
+    @CommandLine.Option(names = {"--sourceroot"},
```

Comment:
We dont support this flag now. The command should be run as `ballerina doc [<project-path>]`

Reply:
You can refer to the run command